src/ComparedQueryable/NativeQueryable/EnumerableRewriter.cs

```diff
@@ -54,13 +56,13 @@ protected override Expression VisitMethodCall(MethodCallExpression m)
             return m;
         }
 
-        private ReadOnlyCollection<Expression> FixupQuotedArgs(MethodInfo mi, ReadOnlyCollection<Expression> argList)
+        protected virtual ReadOnlyCollection<Expression> FixupQuotedArgs(MethodInfo mi, ReadOnlyCollection<Expression> argList)
         {
             ParameterInfo[] pis = mi.GetParameters();
             if (pis.Length > 0)
             {
                 List<Expression> newArgs = null;
-                for (int i = 0, n = pis.Length; i < n; i++)
+                for (int i = 0, n = argList.Count; i < n; i++)

```

Comment:
@herkyg actually, I changed this back again. I got confused - this is actually the crux of the change, though this time, I left more comments in the code explaining why these choices were made.